tkg/managementcomponents/management_component_install.go

```diff
@@ -167,10 +167,28 @@ func NoopDeletePackageInstall(clusterClient clusterclient.Client, addonName, nam
 	return nil
 }
 
+// DeleteLegacyAkoOperatorPackageInstall removes legacy management cluster ako operator packageInstall
+func DeleteLegacyAkoOperatorPackageInstall(clusterClient clusterclient.Client, akoOperatorAddonName string) error {
+	akoOperatorPkgiName := akoOperatorName
+
+	if err := pauseAddonSecretReconciliation(clusterClient, akoOperatorAddonName, constants.TkgNamespace); err != nil {
+		return err
+	}
+
+	if err := pausePackageInstallReconciliation(clusterClient, akoOperatorPkgiName, constants.TkgNamespace); err != nil {
+		return err
+	}
+
+	if err := NoopDeletePackageInstall(clusterClient, akoOperatorPkgiName, constants.TkgNamespace); err != nil {
```

Comment:
May I ask what is the package version of akoo after integrated into tkg-pkg? I am asking this because if the akoo's version becomes same as t-f, then the problem below would happen:
* After delete the old pkgi and let tkg-pkg to install the new pkgi, it may still point to the old akoo package, which is from core package repo. Becase t-f version is 0.2x.x and akoo version was 1.x.x, kapp will treat the old akoo package as the default newer version.
* Core package repo will be removed after tkg-pkg deployed, so the old akoo package will be gone in the end. But the pkgi won't become to point to the new akoo package, because kapp does not allow package downgrade. There will be a ReconcileFailure in the package status.
I suggest to
* follow the addon-manager's behavior to retrieve the version from bom to make sure the new pkgi points to the right version, and expose a dev env param to overwrite the version in case the tkg-pkg is not from bom for developing purpose.
* OR, remain the akoo version instead of following t-f version

Reply:
Do you mean this will still happens even if we changed the packageInstall name and packageName?
The previous package name is ako-operator.tanzu.vmware.com and now it's ako-operator-v2.tanzu.vmware.com.

Reply:
Oh, with the name changed it should be fine, thanks!